### PR TITLE
attributes: update to syn 2.0

### PR DESCRIPTION
**Pull request by davidpdrsn**
## Motivation

syn 2.0 is out!

## Solution

Update to syn 2.0 :rocket:

### CODE REVIEW

**Comment by davidpdrsn** [comment]
Hm syn's [MSRV is now `1.56`](https://github.com/dtolnay/syn/releases/tag/2.0.0)

File: tracing-attributes/src/attr.rs
Comment (jplatte) [inline tracing-attributes/src/attr.rs:242]: Type annotation no longer necessary.

File: tracing-attributes/src/attr.rs
Comment (jplatte) [inline tracing-attributes/src/attr.rs:294]: Same here.

**Comment by hawkw** [comment]
I think it's acceptable to do a MSRV bump to 1.56, since a lot of the ecosystem has done it and it's supported in most Linux distros. Should allow us to clean up some MSRV special-casing anyway.

**Comment by davidbarsky** [comment]
@davidpdrsn https://github.com/tokio-rs/tracing/pull/2546 should fix CI for you; I'll update _this_ PR once #2546 lands.

**Comment by bryantbiggs** [comment]
This would also close  #2543

**Review by hawkw — APPROVED** [review]
looks good to me! will merge when CI passes!

File: tracing-attributes/Cargo.toml
Comment (hawkw) [inline tracing-attributes/Cargo.toml:38]: gah i accidentally deleted the closing `}` while merging `master` :(
```suggestion
syn = { version = "2.0", default-features = false, features = ["full", "parsing", "printing", "visit-mut", "clone-impls", "extra-traits", "proc-macro"] }
```

**Pull request merged**
